packages/atproto-browser/app/actions.ts

```diff
@@ -3,25 +3,86 @@
 import { getAtUriPath } from "@/lib/util";
 import { AtUri, isValidHandle } from "@atproto/syntax";
 import { redirect } from "next/navigation";
+import { parse as parseHtml } from "node-html-parser";
 
 export async function navigateUri(_state: unknown, formData: FormData) {
   const uriInput = formData.get("uri") as string;
+
   const handle = parseHandle(uriInput);
 
   if (handle) {
     redirect(getAtUriPath(new AtUri(`at://${handle}`)));
   }
 
-  let uri;
+  const result =
+    uriInput.startsWith("http://") || uriInput.startsWith("https://")
+      ? await getAtUriFromHttp(uriInput)
+      : parseUri(uriInput);
+
+  if ("error" in result) {
+    return result;
+  }
+
+  redirect(getAtUriPath(result.uri));
+}
+
+type UriParseResult =
+  | {
+      error: string;
+    }
+  | { uri: AtUri };
+
+async function getAtUriFromHttp(url: string): Promise<UriParseResult> {
+  const response = await fetch(url);
+  if (!response.ok) {
+    return { error: `Failed to fetch ${url}` };
+  }
+
+  const linkHeader = response.headers.get("Link");
+  if (linkHeader) {
+    const atUriMatch = linkHeader.match(/<(at:\/\/[^>]+)>; rel="alternate"/);
```

Comment:
This regex is a copilot special, I need to test it (hence the draft state of this PR)